frontend/src/App.tsx

```diff
@@ -283,7 +283,7 @@ export class App extends PureComponent<Props, State> {
           this.handlePageInfoChanged(pageInfo),
         reportFinished: (status: ForwardMsg.ReportFinishedStatus) =>
           this.handleReportFinished(status),
-        uploadReportProgress: (progress: string | number) =>
+        uploadReportProgress: (progress: number) =>
```

Comment:
I am not sure why this is the case. Looks like the person who wrote it no longer works here. @tvst maybe?

Reply:
https://github.com/streamlit/streamlit/blob/develop/proto/streamlit/proto/ForwardMsg.proto#L59 suggests that it is okay.

```
    /**
     * Properties of a ForwardMsg.
     * @exports IForwardMsg
     * @interface IForwardMsg
     * @property {string|null} [hash] ForwardMsg hash
     * @property {IForwardMsgMetadata|null} [metadata] ForwardMsg metadata
     * @property {IInitialize|null} [initialize] ForwardMsg initialize
     * @property {INewReport|null} [newReport] ForwardMsg newReport
     * @property {IDelta|null} [delta] ForwardMsg delta
     * @property {IPageInfo|null} [pageInfoChanged] ForwardMsg pageInfoChanged
     * @property {IPageConfig|null} [pageConfigChanged] ForwardMsg pageConfigChanged
     * @property {ForwardMsg.ReportFinishedStatus|null} [reportFinished] ForwardMsg reportFinished
     * @property {number|null} [uploadReportProgress] ForwardMsg uploadReportProgress
     * @property {string|null} [reportUploaded] ForwardMsg reportUploaded
     * @property {ISessionState|null} [sessionStateChanged] ForwardMsg sessionStateChanged
     * @property {ISessionEvent|null} [sessionEvent] ForwardMsg sessionEvent
     * @property {string|null} [refHash] ForwardMsg refHash
     */
```


https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/report_session.py#L552
https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/storage/file_storage.py#L68
https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/storage/s3_storage.py#L209